### ISSUE TITLE
[MacOS][Sequoia] Proxy error, connectivity issue or a bug

### ISSUE THREAD

**Issue by serbanghita** [metalbear-co/mirrord#2796]
### Bug Description

Mac OS + running Docker image with mirrord inside

```
Mac OS Sequoia 15.0 (24A335)
Docker version 27.2.0, build 3ab4256
mirrord 3.118.1 (the same happens on 3.118.0)
```

### Steps to Reproduce

`Dockerfile` - this is a PHP app, image also installs mirrord

```
{
    "agent": {
        "ephemeral": true
    },
    "feature": {
        "network": {
            "incoming": {
                "ignore_localhost": true
            },
            "outgoing": {
                "ignore_localhost": true,
                "filter": {
                    "local": [
                        ":9000",
                        ":9003"
                    ]
                }
            },
            "dns": {
                "enabled": true,
                "filter": {
                    "local": [
                        ":9000",
                        ":9003"
                    ]
                }
            }
        },
        "env": {
            "override": {
                "DEV_TOOLS_SWAP": "1"
            }
        }
    }
}
```

The way we run mirrord:

```
docker run --entrypoint mirrord -v /var/folders/_l/ybwhx1c14597q3528lv_vrwm0000gn/T/devtools-mirrordHL3wPl/config.json:/mirrord.json -v /var/folders/_l/ybwhx1c14597q3528lv_vrwm0000gn/T/devtools-mirrordunIYCU/config:/root/.kube/config -it --add-host host.docker.internal:host-gateway --rm --env DEV_TOOLS_SWAP=1 -v /Users/serbanghita/work/AAA-core-api:/var/www core-api-swap exec -t deployment/core-api -n serbanghita-core --context AAA-dev --no-telemetry --steal -f /mirrord.json -- env COMPOSER_ALLOW_SUPERUSER=1 start.sh # Finished with exit code 137.
```

### Backtrace

```shell
Proxy error, connectivity issue or a bug.
                    Please report it to us on https://github.com/metalbear-co/mirrord/issues/new?assignees=&labels=bug&projects=&template=bug_report.yml
                    You can find the `mirrord-intproxy` logs in /tmp.
                    connection closed
[03-Oct-2024 11:57:27] WARNING: [pool www] child 58 exited on signal 9 (SIGKILL) after 56.082174 seconds from start
[03-Oct-2024 11:57:27] NOTICE: [pool www] child 109 started
thread '<unnamed>' panicked at mirrord/layer/src/lib.rs:637:14:
failed to establish proxy connection for child: IoFailed(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 239874080
```
```


### mirrord layer logs

```shell
2024-10-03T12:14:13.390822Z  INFO ThreadId(02) mirrord_layer::error: mirrord-layer: IPv6 can't be used with mirrord
2024-10-03T12:14:13.391614Z DEBUG ThreadId(04) check_query: mirrord_layer::socket::dns_selector: new self=DnsSelector { filters: [Port(9000), Port(9003)], filter_is_local: true } node="datadog-apm-agent.infra.svc.cluster.local" port=8126
2024-10-03T12:14:13.391651Z DEBUG ThreadId(04) check_query: mirrord_layer::socket::dns_selector: return=Success(()) self=DnsSelector { filters: [Port(9000), Port(9003)], filter_is_local: true } node="datadog-apm-agent.infra.svc.cluster.local" port=8126
2024-10-03T12:14:13.391670Z DEBUG ThreadId(04) check_query: mirrord_layer::socket::dns_selector: close time.busy=18.8µs time.idle=39.9µs self=DnsSelector { filters: [Port(9000), Port(9003)], filter_is_local: true } node="datadog-apm-agent.infra.svc.cluster.local" port=8126
Proxy error, connectivity issue or a bug.
                    Please report it to us on https://github.com/metalbear-co/mirrord/issues/new?assignees=&labels=bug&projects=&template=bug_report.yml
                    You can find the `mirrord-intproxy` logs in /tmp.
                    connection closed
[03-Oct-2024 12:14:13] WARNING: [pool www] child 67 exited on signal 9 (SIGKILL) after 298.789201 seconds from start
2024-10-03T12:14:13.405388Z DEBUG ThreadId(01) mirrord_layer: Process 45 forking!.
2024-10-03T12:14:13.407564Z DEBUG ThreadId(01) mirrord_layer: Child process id is 159.
[03-Oct-2024 12:14:13] NOTICE: [pool www] child 159 started
2024-10-03T12:14:13.408089Z DEBUG ThreadId(01) mirrord_layer: Child process initializing layer.
thread '<unnamed>' panicked at mirrord/layer/src/lib.rs:637:14:
failed to establish proxy connection for child: IoFailed(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 3271448272
[03-Oct-2024 12:14:13] WARNING: [pool www] child 159 exited on signal 6 (SIGABRT) after 0.003697 seconds from start
2024-10-03T12:14:13.411358Z DEBUG ThreadId(01) mirrord_layer: Process 45 forking!.
2024-10-03T12:14:13.412575Z DEBUG ThreadId(01) mirrord_layer: Child process id is 160.
[03-Oct-2024 12:14:13] NOTICE: [pool www] child 160 started
2024-10-03T12:14:13.412735Z DEBUG ThreadId(01) mirrord_layer: Child process initializing layer.
thread '<unnamed>' panicked at mirrord/layer/src/lib.rs:637:14:
failed to establish proxy connection for child: IoFailed(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 3271448272
```
```


### Internal proxy logs

_No response_

### Your operating system and version

Mac OS Sequoia 15.0 (24A335)

### Local process

docker

### Local process version

_No response_

### Additional Info

_No response_

**Comment by pietro-rutzen** [comment]
Can confirm this on my end as well:
Docker version 27.2.0, build 3ab4256
MacOS 15.0 (24A335)
Mirrord 3.118.0

```thread '<unnamed>' panicked at mirrord/layer/src/lib.rs:631:14:
failed to establish proxy connection for child: IoFailed(Os { code: 111, kind: ConnectionRefused, message: "Connection refused" })
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 3302990656
/usr/local/bin/start.sh: line 50:   123 Aborted                 wait-for-it "${DB_HOST:-mysql}:${DB_PORT:-3307}" -t 200
thread '<unnamed>' panicked at mirrord/layer/src/lib.rs:631:14:

**Comment by serbanghita** [comment]
Magically solved by updating to Mac OS Sequoia `15.0.1`.
I still don't know the root cause of this
